TargetComposition/SymBuilder/SymBuilderTests/BasicTypeTests.js

```diff
@@ -0,0 +1,929 @@
+"use strict";
+//
+// [Harness: run notepad.exe]
+//
+
+//**************************************************************************
+// BasicTypeTests.js
+//
+// Unit tests for basic types within the type builder. 
+//
+// NOTE: The test harness will read the script before opening the engine and having it execute it.
+//       During that process, it will read any comment lines at the beginning of the file (after any
+//       optional "use strict";) and will interpret // [Harness: <something>] as a command to the
+//       test harness.  Here, this script will be executed against "notepad.exe" because of the
+//       harness command.

```

Comment:
I was going to suggest that the metadata comes direct as a data model object, but given that we don't have the data model XML parsing support integrated into DbgX, that's definitely going to be messier right now. If we made this something more generalized, maybe we should consider doing that instead of parsing the script with a regex.

Reply:
That had crossed my mind...  But correct me if I'm mistaken, I didn't think any of the necessary infrastructure to bridge to JavaScript was available until the session was connected...?

The reason I did it this way (at least at the moment) was that the tests need *SOME MECHANISM* of getting that initial session connected (e.g.: start up notepad...  or open a dump file)....

Maybe I'm misremembering things around the timing of spinning up the JS bridge...

Reply:
JS seems to work with no target. I just tested a basic little script and got it to work before initializing any target. You'll get warnings about "the debugger does not have a current process or thread" but basic stuff seems to be fine. You do have to manually load jsprovider because default extensions haven't been loaded yet.

That said, I'd probably need a few small tweaks to DbgX to let you do that. I'm happy to make those tweaks though if you want to do something like that.